syntax = "proto3";

message TransferRequest
{
    string key = 1;             // 固定值 "START-FILE-TRANSFER"
}

message TransferResponse
{
    string key = 1;             // 固定值 "START-FILE-TRANSFER"
    int32 port = 2;             // TCP 端口
}

message StopTransferRequest
{
    string key = 1;             // 固定值 "STOP-FILE-TRANSFER"
}

message StopTransferResponse
{
    string key = 1;             // 固定值 "STOP-FILE-TRANSFER"
}

message SendFileRequest
{
    string file_name = 1;       // 文件名
    int32 parent_id = 2;        // 父目录ID,0表示没有父目录
}

message SendFileResponse
{
    string file_name = 1;       // 文件名
    int32 file_id = 2;          // 文件ID，0为无效id
}

message SendFileBlockRequest
{
    int32 file_id = 1;          // 文件ID
    int64 block_serial = 2;     // 块序号
    int64 block_pos = 3;        // 块起点
    int64 block_size = 4;       // 块大小
    bytes block_data = 5;       // 块数据 
}

message SendFileBlockResponse
{
    int32 file_id = 1;          // 文件ID
    int64 block_serial = 2;     // 块序号
}

message StopSendFileRequest
{
    int32 file_id = 1;          // 文件ID
    string file_sha256 = 2;     // 文件哈希
}

message StopSendFileResponse
{
    string file_name = 1;       // 文件名
    string file_sha256 = 2;     // 文件哈希
    bool correct = 3;           // 是否成功(校验哈希)
}

message SendDirRequest
{
    string dir_name = 1;       // 目录名
    int32 parent_id = 2;       // 父目录ID
}

message SendDirResponse
{
    string dir_name = 1;       // 目录名
    int32 dir_id = 2;          // 目录ID
}

message StopSendDirRequest
{
    string dir_id = 1;          // 目录ID
}

message StopSendDirResponse
{
    int32 dir_id = 1;           // 目录ID
}

message AbortSendFileRequest
{
    int32 file_id = 1;          // 文件ID
}

message AbortSendFileResponse
{
    int32 file_id = 1;          // 文件ID
    bool success = 2;           // 是否成功
}